polygon-digital-carbon/src/utils/CarbonCredit.ts

```diff
@@ -199,6 +201,22 @@ function updateC3Call(tokenAddress: Address, carbonCredit: CarbonCredit): Carbon
   return carbonCredit
 }
 
+function updateCMARKCall(tokenAddress: Address, carbonCredit: CarbonCredit): CarbonCredit {
+  let token = loadOrCreateToken(tokenAddress)
+  let splittedSymbol = token.symbol.split("-")
+  let projectId = splittedSymbol.slice(0,2).join("-")
```

Comment:
if we want to follow the pattern in update<credit>Call we can get these details directly from the contract with tokenAddress + bind and call the appropriate getter.

i don't see any downside to splitting the token symbol to get the projectId from any perspective though so there's no real reason to change it.